src/xkbcomp/symbols.c

```diff
@@ -1152,14 +1152,21 @@ HandleModMapDef(SymbolsInfo *info, ModMapDef *def)
     xkb_mod_index_t ndx;
     bool ok;
     struct xkb_context *ctx = info->ctx;
-
-    ndx = XkbModNameToIndex(&info->mods, def->modifier, MOD_REAL);
-    if (ndx == XKB_MOD_INVALID) {
-        log_err(info->ctx,
-                "Illegal modifier map definition; "
-                "Ignoring map for non-modifier \"%s\"\n",
-                xkb_atom_text(ctx, def->modifier));
-        return false;
+    xkb_atom_t none = xkb_atom_intern(ctx, "None", 4);
+
+    if (def->modifier == none) {
+        // Handle special "None" entry
+        ndx = XKB_MOD_NONE;
```

Comment:
Let's add a comment on the ModMapDef `modifier` field in the struct definition, `// Can also be XKB_MOD_NONE, meaning "don't add a modifier to the modmap".`

Reply:
I think you meant ModMapEntry. I added a comment to both.